packages/uikit-react-native-foundation/src/components/ImageWithPlaceholder/index.tsx

```diff
@@ -43,26 +44,48 @@ type Props = {
   width?: number | string;
   height?: number | string;
   style?: StyleProp<ViewStyle>;
+  cachedSource?: string;
 };
 const ImageWithPlaceholder = (props: Props) => {
   const { palette, select, colors } = useUIKitTheme();
   const [imageNotFound, setImageNotFound] = useState(false);
-
+  const [loading, setLoading] = useState(true);
+  
   const key = useRetry(imageNotFound);
   return (
     <Box
-      style={[{ overflow: 'hidden', width: props.width, height: props.height }, props.style]}
+      style={[{ overflow: 'hidden', width: props.width, height: props.height }, props.style, styles.container]}
       backgroundColor={select({ dark: palette.background400, light: palette.background100 })}
     >
+      {loading && <LoadingSpinner />}
       <Image
         key={key}
         source={props.source}
         style={[StyleSheet.absoluteFill, imageNotFound && styles.hide]}
         resizeMode={'cover'}
         resizeMethod={'resize'}
-        onError={() => setImageNotFound(true)}
-        onLoad={() => setImageNotFound(false)}
+        onError={() => {
+          setLoading(false);
+          setImageNotFound(true);
+        }}
+        onLoad={() => {
+          setLoading(false);
+          setImageNotFound(false);
+        }}
       />
+      {
+        props.cachedSource && (loading || imageNotFound) && (
+          <Image
+            key={`local-${key}`}
+            source={{
+              uri: props.cachedSource,
+            }}
+            style={StyleSheet.absoluteFill}
+            resizeMode={'cover'}
+            resizeMethod={'resize'}
+          />
+        )
+      }
       {imageNotFound && (
```

Comment:
Should this now include `&& !props.cachedSource`?

Reply:
ooo good point